tests/test_multidict.py

```diff
@@ -281,34 +281,79 @@ def test_keys_is_set_less(self, cls: Type[MutableMultiMapping[str]]) -> None:
 
         assert d.keys() < {"key", "key2"}
 
-    def test_keys_is_set_less_equal(self, cls: Type[MutableMultiMapping[str]]) -> None:
-        d = cls([("key", "value1")])
+    @pytest.mark.parametrize(
+        ("contents", "expected"),
+        (
+            ([("key", "value1")], True),
+            ([("key", "value1"), ("key2", "value2")], True),
+            ([("key", "value1"), ("key2", "value2"), ("key3", "value3")], False),
+            ([("key", "value1"), ("key3", "value3")], False),
+        ),
+    )
+    def test_keys_is_set_less_equal(
+        self,
+        cls: Type[MutableMultiMapping[str]],
+        contents: List[Tuple[str, str]],
+        expected: bool,
+    ) -> None:
+        d = cls(contents)
 
-        assert d.keys() <= {"key"}
+        result = d.keys() <= {"key", "key2"}
+        assert result is expected
 
     def test_keys_is_set_equal(self, cls: Type[MutableMultiMapping[str]]) -> None:
         d = cls([("key", "value1")])
 
         assert d.keys() == {"key"}
 
     def test_keys_is_set_greater(self, cls: Type[MutableMultiMapping[str]]) -> None:
-        d = cls([("key", "value1")])
+        d = cls([("key", "value1"), ("key2", "value2")])
 
-        assert {"key", "key2"} > d.keys()
+        assert d.keys() > {"key"}
 
+    @pytest.mark.parametrize(
+        ("set_", "expected"),
+        (
+            ({"key"}, True),
+            ({"key", "key2"}, True),
+            ({"key", "key2", "key3"}, False),
+            ({"key3"}, False),
+        ),
+    )
     def test_keys_is_set_greater_equal(
-        self,
-        cls: Type[MutableMultiMapping[str]],
+        self, cls: Type[MutableMultiMapping[str]], set_: Set[str], expected: bool
+    ) -> None:
+        d = cls([("key", "value1"), ("key2", "value2")])
+
+        result = d.keys() >= set_
+        assert result is expected
+
+    @pytest.mark.parametrize("op", (operator.le, operator.lt, operator.ge, operator.gt))
+    def test_keys_compare_type_error(
+        self, cls: Type[MutableMultiMapping[str]], op: Callable[[object, object], bool]
     ) -> None:
         d = cls([("key", "value1")])
 
-        assert {"key"} >= d.keys()
+        with pytest.raises(
+            TypeError,
+            match=(
+                "^'.+' not supported between instances of '.*_KeysView' and 'NoneType'$"
+            ),
+        ):
+            op(d.keys(), None)
```

Comment:
Could you implement a variant of https://github.com/aio-libs/multidict/pull/936#discussion_r1473671524 instead?

This would end up looking cleaner and not needing a `raises()` check. WDYT?